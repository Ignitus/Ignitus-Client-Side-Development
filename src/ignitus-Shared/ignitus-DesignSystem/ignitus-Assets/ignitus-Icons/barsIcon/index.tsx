import React, { HTMLAttributes } from 'react';

type Props = HTMLAttributes<SVGElement>;

export const BarsIcon = ({ title = 'Bars Icon', ...rest }: Props) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    aria-label={title}
    viewBox="0 0 24 24"
    {...rest}
  >
    <title>{title}</title>
    <path d="M0 0h24v24H0z" fill="none" />
    <path d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
  </svg>
);

/* SVG credits goes to https://material.io/resources/icons/ */
