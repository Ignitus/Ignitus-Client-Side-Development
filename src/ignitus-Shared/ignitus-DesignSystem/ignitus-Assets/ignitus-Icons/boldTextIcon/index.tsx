import React, { HTMLAttributes } from 'react';

type Props = HTMLAttributes<SVGElement>;

export const BoldTextIcon = ({ title = 'BoldText Icon', ...rest }: Props) => (
  <svg
    aria-label={title}
    role="img"
    xmlns="http://www.w3.org/2000/svg"
    width="9" 
    height="12"
    viewBox="0 0 9 12"
    fill="none"
    {...rest}
  >
    <title>{title}</title>
    <path d="M7.2 5.82C8.01209 5.24571 8.5814 4.30286 8.5814 3.42857C8.5814 1.49143 7.11628 0 5.23256 0H0V12H5.89395C7.64372 12 9 10.5429 9 8.75143C9 7.44857 8.28 6.33429 7.2 5.82ZM2.51163 2.14286H5.02326C5.71814 2.14286 6.27907 2.71714 6.27907 3.42857C6.27907 4.14 5.71814 4.71429 5.02326 4.71429H2.51163V2.14286ZM5.44186 9.85714H2.51163V7.28571H5.44186C6.13674 7.28571 6.69767 7.86 6.69767 8.57143C6.69767 9.28286 6.13674 9.85714 5.44186 9.85714Z" fill="white"/>  </svg>
);

