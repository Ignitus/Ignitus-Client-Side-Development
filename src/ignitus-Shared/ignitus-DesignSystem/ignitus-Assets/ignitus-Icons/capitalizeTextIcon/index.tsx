import React, { HTMLAttributes } from 'react';

type Props = HTMLAttributes<SVGElement>;

export const CapitalizeTextIcon = ({
  title = 'CapitalizeText Icon',
  ...rest
}: Props) => (
  <svg
    aria-label={title}
    role="img"
    xmlns="http://www.w3.org/2000/svg"
    width="16"
    height="13"
    viewBox="0 0 16 13"
    fill="none"
    {...rest}
  >
    <title>{title}</title>
    <path
      d="M5.125 1.35419C5.125 2.01127 5.65542 2.54169 6.3125 2.54169H9.08333V10.8542C9.08333 11.5113 9.61375 12.0417 10.2708 12.0417C10.9279 12.0417 11.4583 11.5113 11.4583 10.8542V2.54169H14.2292C14.8863 2.54169 15.4167 2.01127 15.4167 1.35419C15.4167 0.697104 14.8863 0.166687 14.2292 0.166687H6.3125C5.65542 0.166687 5.125 0.697104 5.125 1.35419ZM1.5625 6.50002H2.75V10.8542C2.75 11.5113 3.28042 12.0417 3.9375 12.0417C4.59458 12.0417 5.125 11.5113 5.125 10.8542V6.50002H6.3125C6.96958 6.50002 7.5 5.9696 7.5 5.31252C7.5 4.65544 6.96958 4.12502 6.3125 4.12502H1.5625C0.905417 4.12502 0.375 4.65544 0.375 5.31252C0.375 5.9696 0.905417 6.50002 1.5625 6.50002Z"
      fill="white"
    />
  </svg>
);
