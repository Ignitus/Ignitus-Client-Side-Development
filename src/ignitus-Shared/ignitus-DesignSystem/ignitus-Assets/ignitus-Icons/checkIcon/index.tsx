import React, { HTMLAttributes } from 'react';

type Props = HTMLAttributes<SVGElement>;

export const CheckIcon = ({ title = 'Check Icon', ...rest }: Props) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="23"
    height="17"
    fill="none"
    viewBox="0 0 23 17"
    aria-label={title}
    {...rest}
  >
    <title>{title}</title>
    <path
      // fill="#fff"
      d="M7.291 13.499l-4.52-4.521a1.27 1.27 0 00-1.81 0 1.27 1.27 0 000 1.808l5.413 5.412a1.286 1.286 0 001.821 0L21.887 2.52a1.27 1.27 0 000-1.809 1.27 1.27 0 00-1.809 0L7.291 13.5z"
    />
  </svg>
);
