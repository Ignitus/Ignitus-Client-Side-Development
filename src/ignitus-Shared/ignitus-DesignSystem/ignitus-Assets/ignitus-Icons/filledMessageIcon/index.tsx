import React, { HTMLAttributes } from 'react';

type Props = HTMLAttributes<SVGElement>;

export const FilledMessageIcon = ({
  title = 'Filled Message Icon',
  ...rest
}: Props) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="28"
    height="22"
    fill="none"
    viewBox="0 0 28 22"
    {...rest}
  >
    <title> Filled Message Icon </title>
    <path
      fill="#006"
      fillRule="evenodd"
      d="M3.334.333h21.333c1.467 0 2.667 1.2 2.667 2.667v16c0 1.467-1.2 2.667-2.667 2.667H3.334A2.674 2.674 0 01.667 19V3c0-1.466 1.2-2.667 2.667-2.667zm12.08 11.12L24.134 6a1.14 1.14 0 00.533-.96c0-.893-.973-1.427-1.733-.96L14 9.667 5.067 4.08c-.76-.467-1.733.067-1.733.96 0 .387.2.747.533.96l8.72 5.454c.867.546 1.96.546 2.827 0z"
      clipRule="evenodd"
    />
  </svg>
);
