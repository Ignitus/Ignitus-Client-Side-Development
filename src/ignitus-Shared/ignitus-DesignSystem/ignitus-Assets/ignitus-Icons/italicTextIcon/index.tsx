import React, { HTMLAttributes } from 'react';

type Props = HTMLAttributes<SVGElement>;

export const ItalicTextIcon = ({ title = 'ItalicText Icon', ...rest }: Props) => (
  <svg
    aria-label={title}
    role="img"
    xmlns="http://www.w3.org/2000/svg"
    width="10" 
    height="12"
    viewBox="0 0 10 12"
    fill="none"
    {...rest}
  >
    <title>{title}</title>
    <path d="M3.33333 0V2.57143H5.175L2.325 9.42857H0V12H6.66667V9.42857H4.825L7.675 2.57143H10V0H3.33333Z" fill="white"/>
    </svg>
);








