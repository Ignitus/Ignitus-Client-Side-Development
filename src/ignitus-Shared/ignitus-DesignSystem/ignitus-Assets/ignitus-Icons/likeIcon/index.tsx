import React, { HTMLAttributes } from 'react';
import { IgnitusBlue } from '../../../ignitus-Atoms/colors';

type Props = HTMLAttributes<SVGElement>;

export const LikeIcon = ({ title = 'Like Icon', ...rest }: Props) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 9 9"
    aria-label={title}
    {...rest}
  >
    <title>{title}</title>
    <path d="M0 0h24v24H0V0z" fill="none" />
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M6.1875 1.125C5.535 1.125 4.90875 1.42875 4.5 1.90875C4.09125 1.42875 3.465 1.125 2.8125 1.125C1.6575 1.125 0.75 2.0325 0.75 3.1875C0.75 4.605 2.025 5.76 3.95625 7.515L4.5 8.00625L5.04375 7.51125C6.975 5.76 8.25 4.605 8.25 3.1875C8.25 2.0325 7.3425 1.125 6.1875 1.125ZM4.5375 6.95625L4.5 6.99375L4.4625 6.95625C2.6775 5.34 1.5 4.27125 1.5 3.1875C1.5 2.4375 2.0625 1.875 2.8125 1.875C3.39 1.875 3.9525 2.24625 4.15125 2.76H4.8525C5.0475 2.24625 5.61 1.875 6.1875 1.875C6.9375 1.875 7.5 2.4375 7.5 3.1875C7.5 4.27125 6.3225 5.34 4.5375 6.95625Z"
      fill={IgnitusBlue}
    />{' '}
  </svg>
);

/* SVG credits goes to https://material.io/resources/icons/ */
