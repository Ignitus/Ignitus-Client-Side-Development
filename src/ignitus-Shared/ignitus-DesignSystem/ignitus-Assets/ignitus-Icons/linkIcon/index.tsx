import React, { HTMLAttributes } from 'react';

type Props = HTMLAttributes<SVGElement>;

export const LinkIcon = ({ title = 'Link Icon', ...rest }: Props) => (
  <svg
    aria-label={title}
    role="img"
    xmlns="http://www.w3.org/2000/svg"
    width="17" 
    height="9"
    viewBox="0 0 17 9"
    fill="none"
    {...rest}
  >
    <title>{title}</title>
    <path d="M1.66143 3.942C1.89946 2.619 3.07263 1.71 4.34782 1.71H6.83868C7.28074 1.71 7.6463 1.323 7.6463 0.855C7.6463 0.387 7.28074 0 6.83868 0H4.43283C2.21401 0 0.233221 1.719 0.0206903 4.059C-0.225845 6.741 1.76344 9 4.2458 9H6.83868C7.28074 9 7.6463 8.613 7.6463 8.145C7.6463 7.677 7.28074 7.29 6.83868 7.29H4.2458C2.62207 7.29 1.33838 5.724 1.66143 3.942ZM5.94605 5.4H11.0468C11.5144 5.4 11.8969 4.995 11.8969 4.5C11.8969 4.005 11.5144 3.6 11.0468 3.6H5.94605C5.47848 3.6 5.09593 4.005 5.09593 4.5C5.09593 4.995 5.47848 5.4 5.94605 5.4ZM12.56 0H10.1542C9.7121 0 9.34654 0.387 9.34654 0.855C9.34654 1.323 9.7121 1.71 10.1542 1.71H12.645C13.9202 1.71 15.0934 2.619 15.3314 3.942C15.6545 5.724 14.3708 7.29 12.747 7.29H10.1542C9.7121 7.29 9.34654 7.677 9.34654 8.145C9.34654 8.613 9.7121 9 10.1542 9H12.747C15.2294 9 17.2187 6.741 16.9806 4.059C16.7681 1.719 14.7788 0 12.56 0Z" fill="white"/>
    </svg>
);


