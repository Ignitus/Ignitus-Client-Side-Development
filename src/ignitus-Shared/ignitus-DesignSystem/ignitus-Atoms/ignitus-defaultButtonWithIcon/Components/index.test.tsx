import React from 'react';
import { mount, render } from 'enzyme';

import { AppIcon } from '../../../../types/iconsTypes/iconEnums';
import { DefaultButtonWithIcon } from './index';

import { AccountCircleIcon } from '../../../ignitus-Assets/ignitus-Icons/accountCircleIcon';
import { AngelListIcon } from '../../../ignitus-Assets/ignitus-Icons/angelListIcon';

const props = {
  size: '50px',
  category: 'None',
  name: AppIcon.AccountCircleIcon,
  content: 'Content',
};

it('should rendered', () => {
  const tree = render(<DefaultButtonWithIcon {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<DefaultButtonWithIcon />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<DefaultButtonWithIcon {...props} />);
  });

  it('should render required html tag(s)', () => {
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('button svg').length).toBe(1);
  });

  it('should render given icon', () => {
    expect(wrapper.find(AccountCircleIcon).length).toBe(1);

    wrapper = mount(
      <DefaultButtonWithIcon {...{ ...props, name: AppIcon.AngelListIcon }} />,
    );

    expect(wrapper.find(AngelListIcon).length).toBe(1);
  });
});
