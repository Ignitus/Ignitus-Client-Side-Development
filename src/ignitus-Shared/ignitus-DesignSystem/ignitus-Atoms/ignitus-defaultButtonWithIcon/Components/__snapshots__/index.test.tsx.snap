// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
<button
  class="css-r6j84o"
>
  Content 
  <svg
    aria-label="AccountCircleIcon"
    category="None"
    class="css-17hpzoo"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      AccountCircleIcon
    </title>
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
    />
  </svg>
</button>
`;
